xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-ui/src/main/resources/CKEditor/Translations.xml

```diff
@@ -464,16 +464,22 @@ ckeditor.plugin.image.imageSelector.documentTreeTab.title=Document Tree
 */
 define(['ckeditor'], function(ckeditor) {
   const translations = $jsontool.serialize($translations);
-  const locale = $jsontool.serialize($xcontext.locale.toString());
+  // CKEditor uses '-' (dash) as locale separator.
+  const locale = $jsontool.serialize($xcontext.locale.toString().toLowerCase().replace('_', '-'));
   for (const [pluginName, pluginTranslations] of Object.entries(translations)) {
     if (ckeditor.plugins.get(pluginName)) {
       // The specified plugin is already defined.
-      ckeditor.plugins.setLang(pluginName, locale, pluginTranslations);
+      // The CKEditor's lang plugin's detect function gives us the locale code currently
+      // used by ckeditor, in case CKEditor falls back to an other one, we still want
+      // to use the locale code initally configured.
+      ckeditor.plugins.setLang(pluginName, ckeditor.lang.detect(), pluginTranslations);
```

Comment:
The comment is not very clear. You say "we still want to use the locale code initially configured" but you don't do this, you use instead ``ckeditor.lang.detect()`` which you say "gives us the locale code currently used by CKEditor" (that could be a fallback, so different from the initially configured locale).